identity_iota/examples/credential.rs

```diff
@@ -0,0 +1,154 @@
+//!
+//! cargo run --example credential
+use identity_core::{
+    common::{AsJson as _, Context, Timestamp},
+    did::DID,
+    key::PublicKey,
+    object,
+    vc::{Credential, CredentialBuilder, CredentialSubject, CredentialSubjectBuilder},
+};
+use identity_crypto::KeyPair;
+use identity_iota::{
+    client::{Client, ClientBuilder, PublishDocumentResponse, TransactionPrinter},
+    did::{IotaDID, IotaDocument},
+    error::Result,
+    helpers::create_ed25519_key,
+    network::Network,
+};
+use identity_proof::{HasProof, LdSignature, SignatureOptions};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct VerifiableCredential {
+    #[serde(flatten)]
+    credential: Credential,
+    proof: LdSignature,
+}
+
+impl HasProof for VerifiableCredential {
+    fn proof(&self) -> &LdSignature {
+        &self.proof
+    }
+
+    fn proof_mut(&mut self) -> &mut LdSignature {
+        &mut self.proof
+    }
+}
+
+#[derive(Debug)]
+struct User {
+    doc: IotaDocument,
+    key: KeyPair,
+    name: String,
+}
+
+impl User {
+    async fn new(name: impl Into<String>, client: &Client) -> Result<Self> {
+        // Generate a DID/keypair
+        let (mut did, key): (IotaDID, KeyPair) = IotaDID::generate_ed25519()?;
+
+        did.id_segments.insert(0, "com".into());
+
+        // Create a signed DID document
+        let pkey: PublicKey = create_ed25519_key(&did, key.public().as_ref())?;
```

Comment:
What does this do? Format it? Either needs a dedicated comment or new name. We have generate ed25519 and create ed25519, how do they differ?

Reply:
This creates a verification method object - the name isn't great.

After https://github.com/iotaledger/identity.rs/pull/64 we'll have `IotaDocument::generate_ed25519` (or similar) and this function will be removed.

I'll fix the comment since it also seems to be wrong and the function will be removed soon